### DISCUSSION TITLE
Torch Swap rejected (4.5.3)

### DISCUSSION THREAD

**Discussion by BenAvermate** [Pryaxis/TShock#2343] [Q&A] [answered]
I updated my server to 4.5.3 and now everytime a player tries to place a torch with torch swap the bouncer rejects it.
```
Bouncer / OnTileEdit rejected  from (placestyle) <user> PlaceTile 4 placeStyle: x expectedStyle: 0
```
I have disabled PreventInvalidPlaceStyle but it still won't allow torch swap.

Am I missing something?

**Comment by Kojirremer** [comment]
Can't confirm myself, but I'll try asking someone who is online on my server to confirm.

**Comment by bartico6** [comment]
Hey @BenAvermate, please post an [issue](https://github.com/Pryaxis/TShock/issues/new/choose) about this so that it can be tracked properly and subsequently fixed - it appears to be related to style checks (torch swap makes you place torches of type not matching the held item which for 99% of placement situations indicates a cheat and should be rejected - torch god buff is something we don't appear to account for) - I have some preeliminary idea of why this happens and how to fix it, but having an issue that I can reference in a PR would be great :)

**Reply by BenAvermate** [reply]
I submitted an issue, thank you for your help!

**Comment by BenAvermate** [comment, marked as answer]
Here is the issue:
https://github.com/Pryaxis/TShock/issues/2344#issue-899656477

**Reply by bartico6** [reply]
Thank you very much, I'll lock this discussion and any further conversation about the problem will happen in the issue since it's very obviously a bug rather than a Q&A style conversation :-)
